src/SiteAlias/ProcessManager.php

```diff
@@ -0,0 +1,112 @@
+<?php
+namespace Drush\SiteAlias;
+
+use Consolidation\SiteProcess\ProcessManager as ConsolidationProcessManager;
+
+use Psr\Log\LoggerInterface;
+use Consolidation\SiteAlias\AliasRecord;
+use Consolidation\SiteProcess\Factory\TransportFactoryInterface;
+use Symfony\Component\Process\Process;
+use Drush\Drush;
+use Drush\Style\DrushStyle;
+use Consolidation\SiteProcess\ProcessBase;
+use Consolidation\SiteProcess\SiteProcess;
+
+/**
+ * The Drush ProcessManager adds a few Drush-specific service methods.
+ */
+class ProcessManager extends ConsolidationProcessManager
+{
+    /**
+     * Run a Drush command on a site alias (or @self).
+     *
+     * @param AliasRecord $siteAlias
+     * @param string $command
+     * @param array $args
+     * @param array $options
+     * @param array $options_double_dash
+     * @return SiteProcess
+     */
+    public function drush(AliasRecord $siteAlias, $command, $args = [], $options = [], $options_double_dash = [])
+    {
+        array_unshift($args, $command);
+        return $this->drushSiteProcess($siteAlias, $args, $options, $options_double_dash);
+    }
+
+    /**
+     * drushSiteProcess should be avoided in favor of the drush method above.
+     * drushSiteProcess exists specifically for use by the RedispatchHook,
+     * which does not have specific knowledge about which argument is the command.
+     *
+     * @param AliasRecord $siteAlias
+     * @param array $args
+     * @param array $options
+     * @param array $options_double_dash
+     * @return ProcessBase
+     */
+    public function drushSiteProcess(AliasRecord $siteAlias, $args = [], $options = [], $options_double_dash = [])
+    {
+        // TODO: If local, we should try to find vendor/bin/drush at the local root
+        // and use that if it exists, falling back to Drush::drushScript() if it does not.
+        $defaultDrushScript = !$siteAlias->isLocal() ? 'drush' : Drush::drushScript();
+
+        // Fill in the root and URI from the site alias, if the caller
+        // did not already provide them in $options.
+        if ($siteAlias->has('uri')) {
+            $options += [ 'uri' => $siteAlias->uri(), ];
+        }
+        if ($siteAlias->hasRoot()) {
+            $options += [ 'root' => $siteAlias->root(), ];
+        }
+        array_unshift($args, $siteAlias->get('paths.drush-script', $defaultDrushScript));
+
+        return $this->siteProcess($siteAlias, $args, $options, $options_double_dash);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * Use ProcessManager::drush() instead of this method when calling Drush.
```

Comment:
This should be Drush::drush() still.

Reply:
If a class can inject a reference to the process manager, then it should prefer this method over Drush::drush(). Both will be available, though, so I'll add Drush::drush() back to the comment.